source/slang/glsl.meta.slang

```diff
@@ -4615,4 +4574,607 @@ __spirv_version(1.3) [require(spirv)]
+__glsl_version(430) [require(glsl)]
+[ForceInline] public $(item.name) atomicAnd(inout $(item.name) mem, $(item.name) data)
+{
+    typeRequireChecks_atomic_using_float0_teir<$(item.name)>();
+    typeRequireChecks_atomic_using_Logical_CAS<$(item.name)>();
+    __target_switch
+    {
+    case glsl:
+    {
+        __intrinsic_asm "atomicAnd($0, $1)";
+    }
+    case spirv:
+        return spirv_asm
+        {
+            OpAtomicAnd $$$(item.name) result &mem Device UniformMemory $data
+        };
+    }
+}
+
+__spirv_version(1.0)[require(spirv)]
+__glsl_version(430) [require(glsl)]
+[ForceInline] public $(item.name) atomicOr(inout $(item.name) mem, $(item.name) data)
+{
+    typeRequireChecks_atomic_using_float0_teir<$(item.name)>();
+    typeRequireChecks_atomic_using_Logical_CAS<$(item.name)>();
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "atomicOr($0, $1)";
+    case spirv:
+        return spirv_asm
+        {
+            OpAtomicOr $$$(item.name) result &mem Device UniformMemory $data
+        };
+    }
+}
+
+__spirv_version(1.0)[require(spirv)]
+__glsl_version(430) [require(glsl)]
+[ForceInline] public $(item.name) atomicXor(inout $(item.name) mem, $(item.name) data)
+{
+    typeRequireChecks_atomic_using_float0_teir<$(item.name)>();
+    typeRequireChecks_atomic_using_Logical_CAS<$(item.name)>();
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "atomicXor($0, $1)";
+    case spirv:
+        return spirv_asm
+        {
+            OpAtomicXor $$$(item.name) result &mem Device UniformMemory $data
+        };
+    }
+}
+
+__spirv_version(1.0)[require(spirv)]
+__glsl_version(430) [require(glsl)]
+[ForceInline] public $(item.name) atomicCompSwap(inout $(item.name) mem, $(item.name) compare, $(item.name) data)
+{
+    typeRequireChecks_atomic_using_float0_teir<$(item.name)>();
+    typeRequireChecks_atomic_using_Logical_CAS<$(item.name)>();
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "atomicCompSwap($0, $1, $2)";
+    case spirv:
+        return spirv_asm
+        {
+            result:$$$(item.name) = OpAtomicCompareExchange &mem Device None None $data $compare
+        };
+    }
+}
+
+${{{{
+}
+}}}}
+
+// all atomic_uint functions are HEAVILY mangled at compile time, 
+// all types are converted into 'inout uint' to the referenced 
+// layout(offset) of the related inputted atomic_uint
+__spirv_version(1.0)[require(spirv)]
+__glsl_version(430) [require(glsl)]
+[ForceInline] public uint atomicCounterIncrement(atomic_uint c)
+{
+
+    __target_switch
+    {
+    case glsl: __intrinsic_asm "atomicAdd($0, 1)";
+    case spirv:
+    {
+        return spirv_asm
+        {
+            OpAtomicIIncrement $$uint result $c Device UniformMemory
+        };
+    }
+    }
+}
+
+__glsl_version(430) [require(glsl)]
+[ForceInline] public uint atomicCounterDecrement_GLSL_wrapper(atomic_uint c)
+{
+    __target_switch
+    {
+    case glsl:
+    {
+        __intrinsic_asm "atomicExchange($0,$0-1)";
+    }
```

Comment:
Do you need __target_switch when it requires only glsl?

And this function is used only once.
It will be better to place near where it is called from.

And the function doesn't "wrap" anything.
I would say "_helper".

Reply:
I need the function since __intrinsic_asm returns (and we must return using atomicCounter to get the decrement logic correct)

I will rename the '_wrapper' to '_helper'